integration-cli/docker_cli_swarm_unix_test.go

```diff
@@ -50,3 +51,62 @@ func (s *DockerSwarmSuite) TestSwarmVolumePlugin(c *check.C) {
 	c.Assert(mounts[0].Name, checker.Equals, "my-volume")
 	c.Assert(mounts[0].Driver, checker.Equals, "customvolumedriver")
 }
+
+// Test network plugin filter in swarm
+func (s *DockerSwarmSuite) TestSwarmNetworkPluginV2(c *check.C) {
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, false)
+
+	// install plugin on d1 and d2
+	pluginName := "aragunathan/g3:latest"
+
+	_, err := d1.Cmd("plugin", "install", pluginName, "--grant-all-permissions")
+	c.Assert(err, checker.IsNil)
+
+	_, err = d2.Cmd("plugin", "install", pluginName, "--grant-all-permissions")
+	c.Assert(err, checker.IsNil)
+
+	// Wait for swarmkit's agent.nodeUpdatePeriod to ensure that the manager knows
+	// about installed plugins.
+	time.Sleep(20 * time.Second)
+
+	// create network
+	networkName := "globalnet"
+	_, err = d1.Cmd("network", "create", "--driver", pluginName, networkName)
+	c.Assert(err, checker.IsNil)
+
+	// create service
+	networks := []swarm.NetworkAttachmentConfig{
+		{Target: networkName, Aliases: nil}}
+	d1.CreateService(c, simpleTestService, setNetworks(networks), setInstances(8))
+
+	// wait for tasks ready
+	waitAndAssert(c, defaultReconciliationTimeout, d1.CheckActiveContainerCount, checker.GreaterThan, 0)
+	waitAndAssert(c, defaultReconciliationTimeout, d2.CheckActiveContainerCount, checker.GreaterThan, 0)
+	waitAndAssert(c, defaultReconciliationTimeout, reducedCheck(sumAsIntegers, d1.CheckActiveContainerCount, d2.CheckActiveContainerCount), checker.Equals, 8)
+
+	// scale down number of instances to 0
+	_, err = d1.Cmd("service", "scale", "top=0")
```

Comment:
nit: `CreateService` returns id that could be used instead to avoid this code breaking when `simpleTestService` is updated.